src/services/refactors/extractSymbol.ts

```diff
@@ -606,13 +628,18 @@ namespace ts.refactor.extractSymbol {
     function collectEnclosingScopes(range: TargetRange): Scope[] {
         let current: Node = isReadonlyArray(range.range) ? first(range.range) : range.range;
         if (range.facts & RangeFacts.UsesThis) {
-            // if range uses this as keyword or as type inside the class then it can only be extracted to a method of the containing class
-            const containingClass = getContainingClass(current);
-            if (containingClass) {
-                const containingFunction = findAncestor(current, isFunctionLikeDeclaration);
-                return containingFunction
-                    ? [containingFunction, containingClass]
-                    : [containingClass];
+            if (range.facts & RangeFacts.UsesThisInFunction) {
+                // fall through
+            }
+            else {

```

Comment:
This is a little unusual (I've only seen `// fall through` for unbroken switch case blocks), how about:

```suggestion
            if ((range.facts & RangeFacts.UsesThisInFunction) !== 0) {
```

Reply:
Or, move the above change up into the first `if`.

Reply:
updated